client/src/app/containers/Home/ActiveGamesTable.jsx

```diff
@@ -0,0 +1,119 @@
+import { useInjectReducer, useInjectSaga } from 'redux-injectors';
+import { selectCurrentUser } from '../../../auth/selectors';
+import { authSaga } from '../../../auth/saga';
+import {
+  sliceKey as authSliceKey,
+  reducer as authReducer,
+} from '../../../auth/slice';
+import { selectActiveGames } from '../../../home/selectors';
+import { homeSaga } from '../../../home/saga';
+import {
+  sliceKey as homeSliceKey,
+  reducer as homeReducer,
+  actions as homeActions,
+} from '../../../home/slice';
+
+const ActiveGamesTable = () => {
+  useInjectReducer({ key: authSliceKey, reducer: authReducer });
+  useInjectSaga({ key: authSliceKey, saga: authSaga });
+  useInjectReducer({ key: homeSliceKey, reducer: homeReducer });
+  useInjectSaga({ key: homeSliceKey, saga: homeSaga });
+  const dispatch = useDispatch();
+  const currentUser = useSelector(selectCurrentUser);
+  const activeGames = useSelector(selectActiveGames(currentUser.id));
+
+  // event handlers
+  const onRefreshActiveGames = () => {
+    dispatch(homeActions.refreshActiveGames({ id: currentUser.id }));
+  };
+
+  let gameButtons = [];
+  if (activeGames) {
+    for (const [gID, activeGame] of Object.entries(activeGames)) {
+      if (!gID || !activeGame) {
+        continue;
+      }
+
+      let opponents = [];
+      for (const [pID, pName] of Object.entries(activeGame.players)) {
+        if (pID === currentUser.id) {
+          continue;
+        }
+        opponents.push(pName);
+      }
+
+      gameButtons.push(
+        <tr>
+          <td class='px-6 py-4 whitespace-no-wrap border-b border-gray-200'>
+            {opponents.join(', ')}
+          </td>
+          <td class='px-6 py-4 whitespace-no-wrap border-b border-gray-200 text-sm leading-5 text-gray-500'>
+            {activeGame.colors[currentUser.id]}
+          </td>
+          <td class='px-6 py-4 whitespace-no-wrap border-b border-gray-200 text-sm leading-5 text-gray-500'>
+            {activeGame.created}
+          </td>
+          <td class='px-6 py-4 whitespace-no-wrap border-b border-gray-200 text-sm leading-5 text-gray-500'>
+            {activeGame.lastMove}
+          </td>
+          <td class='px-6 py-4 whitespace-no-wrap text-right border-b border-gray-200 text-sm leading-5 font-medium'>
+            <button key={gID} onClick={() => goToGame({ gID })}>
+              Play!
+            </button>
+          </td>
+        </tr>,
+      );
+    }
+  }
+
+  return (
+    <div>
+      <div class='flex flex-col'>
+        <div class='-my-2 py-2 overflow-x-auto sm:-mx-6 sm:px-6 lg:-mx-8 lg:px-8'>
+          <div class='align-middle inline-block min-w-full shadow overflow-hidden sm:rounded-lg border-b border-gray-200'>
+            <table class='min-w-full'>
+              <thead>
+                <tr>
+                  <th class='px-6 py-3 border-b border-gray-200 bg-gray-50 text-left text-xs leading-4 font-medium text-gray-500 uppercase tracking-wider'>
```

Comment:
These classes are repeated for each `<th>`, can we extract them into a component class in `tailwind.css`?

Reply:
You could do that or use something like https://www.npmjs.com/package/@emotion/styled

Reply:
@amichael497 They use something similar ([styled coomponents, I think](https://github.com/styled-components/styled-components) in the [react boilerplate](https://github.com/react-boilerplate/react-boilerplate); is the main advantage to have a logical separation between styling and the component tree? With tailwind, we're not actually writing any css (or very little css) ourselves since it provides all of the utility classes out of the box.